traffic_ops/traffic_ops_golang/api/generic_crud.go

```diff
@@ -156,6 +166,37 @@ func GenericUpdate(val GenericUpdater) (error, error, int) {
 	return nil, nil, http.StatusOK
 }
 
+// GenericOptionsDelete does a Delete (DELETE) for the given GenericOptionsDeleter object and type. Unlike
+// GenericDelete, there is no requirement that a specific key is used as the parameter.
+// GenericOptionsDeleter.DeleteKeyOptions() specifies which keys can be used.
+func GenericOptionsDelete(val GenericOptionsDeleter) (error, error, int) {
+	where, _, _, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(val.APIInfo().Params, val.DeleteKeyOptions())
+	if len(errs) > 0 {
+		return util.JoinErrs(errs), nil, http.StatusBadRequest
+	}
+
+	query := val.DeleteQueryBase() + where
+	tx := val.APIInfo().Tx
+	result, err := tx.NamedExec(query, queryValues)
+	if err != nil {
+		return nil, errors.New("deleting " + val.GetType() + ": " + err.Error()), http.StatusInternalServerError
```

Comment:
should this be ParseDBError? that parses some postgres errors into non 500 errors

Reply:
Changed to ParseDBError in 2ada94e8f2.